base/base/crc32_s390x.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <crc32-s390x.h>
+
+inline uint32_t s390x_crc32_u8(uint32_t crc, uint8_t v)
+{
+    return crc32c_le_vx(crc, reinterpret_cast<unsigned char *>(&v), sizeof(v));
+}
+
+inline uint32_t s390x_crc32_u16(uint32_t crc, uint16_t v)
+{
+    v = __builtin_bswap16(v);
```

Comment:
Ideally this would be a `std::byteswap`, however, this is unavailable to `base/` as it is compiled with C++20.